op-node/rollup/derive/batch_mux_test.go

```diff
@@ -0,0 +1,68 @@
+package derive
+
+import (
+	"context"
+	"io"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-service/testlog"
+)
+
+func TestBatchMux_LaterHolocene(t *testing.T) {
+	log := testlog.Logger(t, log.LevelTrace)
+	ctx := context.Background()
+	l1A := eth.L1BlockRef{Time: 0, Hash: common.Hash{0xaa}}
+	l1B := eth.L1BlockRef{Time: 12, Hash: common.Hash{0xbb}}
+	cfg := &rollup.Config{
+		HoloceneTime: &l1B.Time,
+	}
+	b := NewBatchMux(log, cfg, nil, nil)
+
+	require.IsType(t, new(BatchQueue), b.SingularBatchProvider)
+
+	err := b.Reset(ctx, l1A, eth.SystemConfig{})
+	require.Equal(t, io.EOF, err)
+	require.IsType(t, new(BatchQueue), b.SingularBatchProvider)
+	require.Equal(t, l1A, b.SingularBatchProvider.(*BatchQueue).origin)
+
+	b.Transform(rollup.Holocene)
+	require.IsType(t, new(BatchStage), b.SingularBatchProvider)
```

Comment:
Similar to the other mux class, probably worth a test for startup when we're already past holocene activation (but not at genesis)

Reply:
See other response, I think this is already covered.